### PR TITLE
[MPLUGIN-431] Remove deprecated items from new maven-plugin-report-plugin

### PR DESCRIPTION
**Pull request by slawekjaranowski**

### CODE REVIEW

File: maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java [hunk 1]
```diff
@@ -107,7 +107,11 @@ else if ( !goalPrefix.equals( defaultGoalPrefix ) )
     static String getDefaultGoalPrefix( MavenProject project )
     {
         String defaultGoalPrefix;
-        if ( "maven-plugin".equalsIgnoreCase( project.getArtifactId() ) )
+        if ( "maven-plugin-report-plugin".equalsIgnoreCase( project.getArtifactId() ) )
+        {
+            defaultGoalPrefix = "plugin-report";
```
Comment (gnodet) [inline maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java:112]: Is that needed ?
I forced at https://github.com/apache/maven-plugin-tools/blob/master/maven-plugin-report-plugin/pom.xml#L182

Comment (slawekjaranowski) [reply maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java:112]: we have warning during build

```
[WARNING] 

Goal prefix is specified as: 'plugin-report'. Maven currently expects it to be 'report'.
```

so I want to mitigate it for next release

Comment (gnodet) [reply maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java:112]: Won't that be the case for all plugins if we start splitting the mojos from the reports ?

Comment (slawekjaranowski) [reply maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java:112]: No - by default `maven` and `plugin` words are removed from artifactId, so it is special case.

Comment (gnodet) [reply maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java:112]: Changed my mind. LGTM

**Review by gnodet — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #738

**Pull request merged**